libdrgn/object.c

```diff
@@ -1168,7 +1168,8 @@ LIBDRGN_PUBLIC struct drgn_error *drgn_object_cmp(const struct drgn_object *lhs,
 						  const struct drgn_object *rhs,
 						  int *ret)
 {
-	const struct drgn_language *lang = &drgn_language_c;
+	// TODO handle the case where LHS and RHS have different types
+	const struct drgn_language *lang = drgn_type_lang(lhs->type);
```

Comment:
I think this is pretty much the sanest way to handle it, unless you have any better ideas.